conda/core/path_actions.py

```diff
@@ -1042,3 +1042,34 @@ def target_full_path(self):
     def __str__(self):
         return ('ExtractPackageAction<source_full_path=%r, target_full_path=%r>'
                 % (self.source_full_path, self.target_full_path))
+
+
+class DumpRecordAction(PathAction):
```

Comment:
I'm not really all that concerned about extra fields at the moment.  Unless as part of this PR you want to nail down the full details of the data model components I sent via email.  That would be just fine too I think.

I think the comment here was really just about getting the right name for the `DumpRepodataRecordAction` class.